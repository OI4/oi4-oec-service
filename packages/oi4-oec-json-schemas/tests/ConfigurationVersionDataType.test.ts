import { matchersWithOptions } from 'jest-json-schema'
import schema from '../schemas/ConfigurationVersionDataType.schema.json'
import uint32 from '../schemas/dataTypes/uint32.schema.json'

import validObjs from './__fixtures__/ConfigurationVersionDataType_valid.json'
import invalidObjs from './__fixtures__/ConfigurationVersionDataType_invalid.json'

expect.extend(
  matchersWithOptions({
    // Loading in a schema which is comprised only of definitions,
    // which means specific test schemas need to be created.
    // This is good for testing specific conditions for definition schemas.
    schemas: [uint32],
    verbose: true,
  })
)

describe('ConfigurationVersionDataType schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it.each(validObjs as [])(
    '(%#) match valid config object to schema -> %s',
    (_name: string, obj) => {
      expect(obj).toMatchSchema(schema)
    }
  )

  it.each(invalidObjs as [])(
    '(%#) match fails for invalid config -> %s',
    (name: string, obj) => {
      expect(obj).not.toMatchSchema(schema)
      expect(() => {
        expect(obj).toMatchSchema(schema)
      }).toThrowErrorMatchingSnapshot(name)
    }
  )
})
