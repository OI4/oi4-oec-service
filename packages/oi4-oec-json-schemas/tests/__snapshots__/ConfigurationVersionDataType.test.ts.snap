// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigurationVersionDataType schema (0) match fails for invalid config -> empty: empty 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'MajorVersion'
    Path:     ConfigurationVersionDataType.schema.json#/required
  must have required property 'MinorVersion'
    Path:     ConfigurationVersionDataType.schema.json#/required
"
`;

exports[`ConfigurationVersionDataType schema (1) match fails for invalid config -> versions out of range: versions out of range 1`] = `
"expect(received).toMatchSchema(schema)

received
  /MajorVersion must be <= 4294967295
    Received: <number> 4294967296
    Path:     ConfigurationVersionDataType.schema.jsondataTypes/uint32.schema.json/maximum
  /MinorVersion must be >= 0
    Received: <number> -4294967295
    Path:     ConfigurationVersionDataType.schema.jsondataTypes/uint32.schema.json/minimum
"
`;
